universal-application-tool-0.0.1/app/repository/ProgramRepository.java

```diff
@@ -46,6 +46,7 @@ public ProgramRepository(
   public Program insertProgramSync(Program program) {
     program.id = null;
     ebeanServer.insert(program);
+    program.refresh();
```

Comment:
question (non-blocking): will the in-memory program not have an ID without this line?

Reply:
It gets the ID, but it doesn't get the version table update.  Something odd in ebeans, I suppose!